streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -128,9 +128,13 @@
     /** <code>client.id</code> */
     public static final String CLIENT_ID_CONFIG = CommonClientConfigs.CLIENT_ID_CONFIG;
 
+    /** <code>rocksdb.config.setter</code> */
     public static final String ROCKSDB_CONFIG_SETTER_CLASS_CONFIG = "rocksdb.config.setter";
     public static final String ROCKSDB_CONFIG_SETTER_CLASS_DOC = "A Rocks DB config setter class that implements the <code>RocksDBConfigSetter</code> interface";
 
+    /** <code>windowstore.changelog.additional.retention.ms</code> */
+    public static final String WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_CONFIG = "windowstore.changelog.additional.retention.ms";
```

Comment:
A meta comment: KAFKA-3595 was filed to allow users to specify different log compaction policies for state stores, while this config is only used for window store. I'm wondering we can provide a more general solution as for KAFKA-3595's scope:

1. In `Stores.StoreFactory` add one more spec after `persistent / inMemory`. More specifically, let `InMemoryKeyValueFactory` and `PersistentKeyValueFactory` to have two functions named `logged()` and `notLogged()` which returns a new `LoggedInMemoryKeyValueFactory` and `LoggedPersistentKeyValueFactory`. `notLogged()` simply set a `logged` flag to false, so that upon `get()` the inner store should not call `enableLogging`. `The `logged` function signature could be defined as:

`LoggedKeyValueFactory<K, V> logged(String cleanupPolicy, Map<String, String> retentionConfigs)` 

where `retentionConfigs` can be used to pass in any server-side log retention / cleaner configs such as `log.retention.ms`, `log.retention.hours`, etc. So that the usage pattern could be `Stores.create("..").withKey().withValue().logged("").persistent()..`. In addition, we can merge in `KTableStoreSupplier` into `LoggedPersistentKeyValueFactory` with `logged` flag to false.

If `logged` or `notLogged` is not called, we can then use the default set of values, where for key-value store it is `compact` with no configs, while for windowed store it is `delete,compact` with `log.retention.ms -> /* retentionMS + 1day */`. What do you think?